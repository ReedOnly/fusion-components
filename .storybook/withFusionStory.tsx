import React from 'react';
import Header from '../src/components/core/Header';
import Content from '../src/components/core/Content';
import Overlay from '../src/components/general/ApplicationGuidance/components/Overlay';

export default (title: string, padding: number = 16) => (stories: () => React.ReactNode) => {
    return (
        <>
            <Overlay fixed>
                <Header aside={null} start={null} content={null} />
            </Overlay>
            <Content>
                <Overlay>
                    <div
                        style={{
                            padding,
                            minHeight: '100%',
                            width: '100%',
                            maxWidth: '100%',
                            boxSizing: 'border-box',
                        }}
                    >
                        {stories()}
                    </div>
                </Overlay>
            </Content>
        </>
    );
};
