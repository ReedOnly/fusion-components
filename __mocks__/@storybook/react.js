import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    // const decoratedStory = api.decorators.reduce((s, decorator) => {
                    //     try {
                    //         return decorator(s, { parameters: api.parameters });
                    //     } catch (e) {
                    //         return s;
                    //     }
                    // }, story);

                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
