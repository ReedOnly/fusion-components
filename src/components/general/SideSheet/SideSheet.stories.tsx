import * as React from 'react';
import { storiesOf } from '@storybook/react';
import withFusionStory from '../../../../.storybook/withFusionStory';
import { ModalSideSheet, SideSheet } from './index';
import { Button } from '@equinor/fusion-components';

const ModalSideSheetStory = () => {
    const [isOpen, setIsOpen] = React.useState(false);

    return (
        <div>
            <Button
                onClick={() => {
                    setIsOpen(true);
                }}
            >
                Open Side Sheet
            </Button>
            <ModalSideSheet
                header="This is the modal side sheet header"
                show={isOpen}
                onClose={() => {
                    setIsOpen(false);
                }}
            >
                <div style={{ padding: '0 32px' }}>This is the modal side sheet content</div>
            </ModalSideSheet>
        </div>
    );
};

const SideSheetStory = () => {
    const [isOpen, setIsOpen] = React.useState(false);

    return (
        <div>
            <Button
                onClick={() => {
                    setIsOpen(true);
                }}
            >
                Open Side Sheet
            </Button>
            <SideSheet
                show={isOpen}
                onClose={() => setIsOpen(false)}
            >
                <div style={{ padding: '0 32px' }}>This is the side sheet content</div>
            </SideSheet>
        </div>
    );
};

storiesOf('General|SideSheet', module)
    .addDecorator(withFusionStory('Side Sheet'))
    .add('Standard', () => <SideSheetStory/>)
    .add('Modal', () => <ModalSideSheetStory />);
