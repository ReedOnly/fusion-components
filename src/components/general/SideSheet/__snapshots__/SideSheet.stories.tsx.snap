// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General|SideSheet Modal 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div>
      <button
        className="container contained primary comfortable"
        disabled={false}
        onClick={[Function]}
        onClickCapture={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <span
          className="button"
        >
          Open Side Sheet
        </span>
      </button>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`General|SideSheet Standard 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div
      style={
        Object {
          "display": "flex",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "flexGrow": 1,
            "padding": 24,
          }
        }
      >
        <p>
          Right, and where am I gonna be? Well, Marty, I want to thank you for all your good advise, I'll never forget it. This is it. This is the answer. It says here that a bolt of lightning is gonna strike the clock tower precisely at 10:04 p.m. next Saturday night. If we could somehow harness this bolt of lightning, channel it into the flux capacitor, it just might work. Next Saturday night, we're sending you back to the future. Hey, George, buddy, you weren't at school, what have you been doing all day? Lynda, first of all, I'm not your answering service. Second of all, somebody named Greg or Craig called you just a little while ago.
        </p>
        <p>
          Right. Thank you. In about thirty years. Whoa, whoa, okay. Next, please. I'm gonna ram him
        </p>
        <p>
          Hey, George, buddy, you weren't at school, what have you been doing all day? Hey beat it, spook, this don't concern you. He's an absolute dream. Check out that four by four. That is hot. Someday, Jennifer, someday. Wouldn't it be great to take that truck up to the lake. Throw a couple of sleeping bags in the back. Lie out under the stars.
        </p>
      </div>
      <div
        className="container comfortable isCollapsed medium"
        style={Object {}}
      >
        <div
          className="header"
        >
          <div
            className="collapseButtonContainer"
          >
            <button
              className="container comfortable"
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="iconContainer"
              >
                <svg
                  height={24}
                  style={
                    Object {
                      "color": "",
                    }
                  }
                  viewBox="0 0 24 24"
                  width={24}
                >
                  <path
                    clipRule="evenodd"
                    d="M7.99878 7C7.99878 6.45 7.54878 6 6.99878 6C6.44878 6 5.99878 6.45 5.99878 7V17C5.99878 17.55 6.44878 18 6.99878 18C7.54878 18 7.99878 17.55 7.99878 17V7ZM13.8188 12L17.6988 15.89C18.0888 16.27 18.0888 16.91 17.7088 17.29C17.3188 17.68 16.6888 17.68 16.2988 17.29L11.7088 12.7C11.3188 12.31 11.3188 11.68 11.7088 11.29L16.2988 6.7C16.4856 6.51275 16.7393 6.40751 17.0038 6.40751C17.2683 6.40751 17.5219 6.51275 17.7088 6.7C18.0988 7.09 18.0988 7.72 17.7088 8.11L13.8188 12Z"
                    fill="currentColor"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
